ffi/capi/src/segmenter_sentence.rs

```diff
@@ -51,6 +67,29 @@ pub mod ffi {
             )?)))
         }
 
+        /// Construct an [`SentenceSegmenter`].
+        #[diplomat::rust_link(
+            icu::segmenter::SentenceSegmenter::try_new_with_options,
+            FnInStruct,
+            hidden
+        )]
+        #[diplomat::attr(supports = fallible_constructors, constructor)]
+        #[diplomat::attr(supports = non_exhaustive_structs, rename = "with_options")]
+        pub fn create_with_options_v1(
+            provider: &DataProvider,
+            options: &SentenceBreakOptionsV1,
+        ) -> Result<Box<SentenceSegmenter>, DataError> {
```

Comment:
Nit: in FFI, since we can't have non-exhaustive structs anyway, when the struct has only 1 field, we usually just make the field be the argument. So, please make this `create_with_content_locale` and the argument can be `&Locale`.

I think we should keep the options bag in the Rust API, just not in the C FFI.